src/domain-engine.js

```diff
@@ -1,16 +1,16 @@
 import MetaDomain from './meta-domain'
 import getRegistration from './get-registration'
 import { get, set, createOrClone } from './utils'
-import { castPath } from './key-path'
+import { castPath, getKeyString } from './key-path'
 
 class DomainEngine {
   /**
    * @param {Microcosm} repo
    */
   constructor(repo) {
-    this.repo = repo
-    this.domains = []
     this.registry = {}
+    this.repo = repo
+    this.domains = [[[], this.repo]]
```

Comment:
just starting to review this PR so i don't know what this is doing or why it's changed but it looks pretty gross